eng/tools/typespec-validation/src/rules/compile.ts

```diff
@@ -14,21 +17,127 @@ export class CompileRule implements Rule {
         stdout.toLowerCase().includes("no output was generated")
       ) {
         success = false;
         errorOutput += "No emitter was configured and/or no output was generated.";
       }
-      if (err) {
-        success = false;
-        errorOutput += err.message;
+
+      if (success) {
+        if (!err) {
+          // Check for *extra* typespec-generated swagger files under the output folder, which
+          // indicates a mismatch between TypeSpec and swaggers.
+
+          // Example 'stdout':
+          //
+          // TypeSpec compiler v0.67.2
+          //
+          // ../resource-manager/Microsoft.Contoso/stable/2021-11-01/contoso.json
+          // ../resource-manager/Microsoft.Contoso/stable/2021-11-01/examples/Operations_List.json
+          //
+          // Compilation completed successfully.
+
+          // Remove ANSI color codes, handle windows and linux line endings
+          const lines = stripAnsi(stdout).split(/\r?\n/);
+
+          // TODO: Use helpers in /.github once they support platform-specific paths
+          const outputSwaggers = lines
+            // Remove leading and trailing whitespace
+            .map((l) => l.trim())
+            // Skip blank lines
+            .filter((l) => l !== "")
+            // Skip first and last lines (version and success messages)
+            .slice(1, -1)
+            // Normalize to platform-specific path
+            .map((l) => normalize(l))
+            // Filter to JSON files
+            .filter((p) => basename(p).toLowerCase().endsWith(".json"))
+            // Exclude examples
+            .filter((p) => !p.split(path.sep).includes("examples"));
+
+          stdOutput += "\nGenerated Swaggers:\n";
+          stdOutput += outputSwaggers.join("\n") + "\n";
+
+          // ../resource-manager/Microsoft.Contoso
+          const outputFolder = dirname(dirname(dirname(outputSwaggers[0])));
+          const outputFilename = basename(outputSwaggers[0]);
+
+          stdOutput += "\nOutput folder:\n";
+          stdOutput += outputFolder + "\n";
+
+          // Filter to only specs matching the folder and filename extracted from the first output-file.
+          // Necessary to handle multi-project specs like keyvault.
+          //
+          // Globby only accepts patterns like posix paths.
+          const pattern = path.posix.join(
+            ...outputFolder.split(path.win32.sep),
+            "**",
+            outputFilename,
+          );
+          const allSwaggers = (await host.globby(pattern, { ignore: ["**/examples/**"] })).map(
+            (p) => normalize(p),
+          );
+
+          // Filter to files generated by TypeSpec
+          const tspGeneratedSwaggers = await filterAsync(
+            allSwaggers,
+            async (swaggerPath: string) => {
+              const swaggerText = await host.readFile(swaggerPath);
+              const swaggerObj = JSON.parse(swaggerText);
+              return swaggerObj["info"]?.["x-typespec-generated"];
```

Comment:
there're some cases where the swagger was generated from cadl 
```suggestion
              return swaggerObj["info"]?.["x-typespec-generated"] || swaggerObj["info"]?.["x-cadl-generated"];
```